internal/nostr/push.go

```diff
@@ -35,24 +92,33 @@ func (svc *Service) NIP47PushNotificationHandler(c echo.Context) error {
 		})
 	}
 
-	if (requestData.WalletPubkey == "") {
+	if requestData.WalletPubkey == "" {
 		return c.JSON(http.StatusBadRequest, ErrorResponse{
 			Message: "wallet pubkey is empty",
 			Error:   "no wallet pubkey in request data",
 		})
 	}
 
-	if (requestData.ConnPubkey == "") {
+	if requestData.ConnPubkey == "" {
 		return c.JSON(http.StatusBadRequest, ErrorResponse{
 			Message: "connection pubkey is empty",
 			Error:   "no connection pubkey in request data",
 		})
 	}
 
+	encryptedPushToken, err := svc.encryptToken(requestData.PushToken)
+	if err != nil {
+		svc.Logger.WithError(err).Error("Failed to encrypt push token")
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "Internal server error",
+			Error:   "failed to encrypt push token",
+		})
+	}
+
 	var existingSubscriptions []Subscription
-	if err := svc.db.Where("push_token = ? AND open = ? AND authors_json->>0 = ? AND tags_json->'p'->>0 = ?", requestData.PushToken, true, requestData.WalletPubkey, requestData.ConnPubkey).Find(&existingSubscriptions).Error; err != nil {
+	if err := svc.db.Where("push_token = ? AND open = ? AND authors_json->>0 = ? AND tags_json->'p'->>0 = ?", encryptedPushToken, true, requestData.WalletPubkey, requestData.ConnPubkey).Find(&existingSubscriptions).Error; err != nil {
 		svc.Logger.WithError(err).WithFields(logrus.Fields{
-			"push_token": requestData.PushToken,
+			"push_token": encryptedPushToken,
```

Comment:
I don't think this should be logged. 
in logs sensitive data is easily leaked. e.g. it is sent through various systems. 
so it's better to not log sensitive data.

Reply:
Done!